x-pack/platform/plugins/shared/lens/public/visualizations/xy/runtime_state/index.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { GeneralDatasourceStates } from '../../../state_management';
+import { XYState } from '../types';
+import { getRuntimeConverters } from './converters';
+
+export function convertToRuntimeState(
+  state: XYState,
+  datasourceStates?: GeneralDatasourceStates
+): XYState {
+  return getRuntimeConverters(datasourceStates).reduce((newState, fn) => fn(newState), state);
```

Comment:
I believe we need to deepClone `state` here, because the `legend_stats` mutates it and we are not passing a cloned state but just a cloned datasourcesStates from the `initialize` method in visualization.tsx.

Reply:
Added `deepClone` to `xy` and `partition` converters. See f879dc96dc8fa7e66e26fccbff709136c52eb521.

Reply:
Also this would mean that we `deepClone` the vis state every time it's initialized. Maybe we can improve this later to separate the check from the transform. Or simply not mutate the state.